pocs/camera/sbigudrv.py

```diff
@@ -347,33 +349,50 @@ def _readout(self, handle, centiseconds, filename, readout_mode_code,
         # Check for the end of the exposure.
         with self._command_lock:
             self._set_handle(handle)
-            self._send_command('CC_QUERY_COMMAND_STATUS', params=query_status_params, results=query_status_results)
+            self._send_command('CC_QUERY_COMMAND_STATUS',
+                               params=query_status_params,
+                               results=query_status_results)
 
         # Poll if needed.
         while query_status_results.status != status_codes['CS_INTEGRATION_COMPLETE']:
             self.logger.debug('Waiting for exposure on {} to complete'.format(handle))
             time.sleep(0.1)
             with self._command_lock:
                 self._set_handle(handle)
-                self._send_command('CC_QUERY_COMMAND_STATUS', params=query_status_params, results=query_status_results)
+                self._send_command('CC_QUERY_COMMAND_STATUS',
+                                   params=query_status_params,
+                                   results=query_status_results)
 
         self.logger.debug('Exposure on {} complete'.format(handle))
 
         # Readout data
         with self._command_lock:
+            self._set_handle(handle)
+            self._send_command('CC_END_EXPOSURE', params=end_exposure_params)
+            self._send_command('CC_START_READOUT', params=start_readout_params)
+            for i in range(height):
+                try:
+                    self._send_command('CC_READOUT_LINE',
+                                       params=readout_line_params,
+                                       results=as_ctypes(image_data[i]))
+                except RuntimeError as err:
+                    message = 'Readout error on {}: expected {} rows, got{}!'.format(handle,
```

Comment:
Does `handle` show the camera id or name? I can't remember but it seems like it tell us which camera it was os that we had to track it down.

Reply:
Neither, unfortunately. It's the internal identifier that the driver defines and uses to identify which camera its talking to.  I could (and probably should) modify these to give the camera UID instead, which is information that the driver can relatively easily retrieve.